pkg/plugins/golang/declarative/v1/internal/templates/dockerfile.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+)
+
+var _ machinery.Template = &Dockerfile{}
+
+// Dockerfile scaffolds a file that defines the containerized build process
+type Dockerfile struct {
+	machinery.TemplateMixin
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *Dockerfile) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = "Dockerfile"
+	}
+
+	f.TemplateBody = dockerfileTemplate
+
+	f.IfExistsAction = machinery.OverwriteFile
+
+	return nil
+}
+
+const dockerfileTemplate = `# Build the manager binary
+FROM golang:1.16 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+
+# Stage channels and make readable
+COPY channels/ /channels/
+RUN chmod -R a+rx /channels/
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /
+COPY --from=builder /workspace/manager .
+COPY --from=builder /channels /channels
+
+USER 65532:65532
+
+ENTRYPOINT ["/manager"]
+`
```

Comment:
Why should we start to scaffold the dockerfile for the declarative one?
Should not these changes be applied for all plugins? 
Why the changes in the dockerfile are valid only for the declarative plugin? 

It shows that your suggestions should be done in: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/scaffolds/internal/templates/dockerfile.go instead

Reply:
This PR is to address the changes required in the original PR https://github.com/kubernetes-sigs/kubebuilder/pull/2318 which adds the `init` command. Should we add the `init` command subcommand for the declarative plugin? Also, should this change be added to every plugin ?